src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -274,20 +275,20 @@ public String getDisplayFunctionName() {
 
     /**
      * Returns colored orb that represents the current state of this node.
-     *
-     * TODO: this makes me wonder if we should support other colored states,
-     * like unstable and aborted --- seems useful.
      */
     @Exported
     public BallColor getIconColor() {
         ErrorAction error = getError();
         BallColor c = null;
         if(error != null) {
             if(error.getError() instanceof FlowInterruptedException) {
+                // TODO: Use FlowInterruptedException.getResult() to determine ball color.
                 c = BallColor.ABORTED;
             } else {
                 c = BallColor.RED;
             }
+        } else if (getPersistentAction(WarningAction.class) != null) {
+            c = BallColor.YELLOW;
```

Comment:
Here is an example of what this change looks like for the following pipeline:

```
stage('unstable-one') {
    unstable('foobar')
}
stage('success') {
    echo('whatever')
}
stage('failure') {
    unstable('three')
    error('uncaught')
}
```
<img width="1068" alt="Screen Shot 2019-04-23 at 12 24 36" src="https://user-images.githubusercontent.com/1068968/56601084-4f4fb880-65c8-11e9-8411-0f5066dd0033.png">

For now only the actual `FlowNode` that has the `WarningAction` attached will be shown as unstable (unlike Blue Ocean in https://github.com/jenkinsci/pipeline-graph-analysis-plugin/pull/25 which searches all `FlowNodes` in a stage/branch for a `WarningAction`.). This seems like the most obvious behavior to me, but I could make it work like Blue Ocean if desired.